FinSearchUnified/Bundle/SearchBundleFindologic/ResponseParser/ResponseParser.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace FinSearchUnified\Bundle\SearchBundleFindologic\ResponseParser;
+
+use FINDOLOGIC\Api\Responses\Response;
+use FINDOLOGIC\Api\Responses\Xml21\Xml21Response;
+use FinSearchUnified\Bundle\SearchBundleFindologic\ResponseParser\QueryInfoMessage\QueryInfoMessage;
+use InvalidArgumentException;
+
+abstract class ResponseParser
+{
+    /**
+     * @var Xml21Response
+     */
+    protected $response;
+
+    public function __construct(Response $response)
+    {
+        $this->response = $response;
+    }
+
+    /**
+     * @return Xml21ResponseParser
```

Comment:
This may not be true. Please add `self`/`ResponseParser` as return typehint. This way we can add new response parsers, without breaking stuff, because they always expect an `Xml21ResponseParser`.